### ISSUE TITLE
 redis for default websocket driver  error

### ISSUE THREAD

**Issue by zhongjq** [swooletw/laravel-swoole#308]
Make sure you read **Issues Guideline** and answer these questions before submitting your issue. Thanks!
(Any **non-English** issues will be closed immediately.)

1. Please provide your PHP and Swoole version. (`php -v` and `php --ri swoole`)
php 7.3.6
swoole  4.3.3

2. Please provide your Laravel/Lumen version.
laravel 5.7

3. Which release version of this package are you using?
v2.6.63

4. What did you do? If possible, provide a recipe for reproducing the error.

I use the redis for default websocket driver，There are occasional log errors in the production environment. 
[reference](http://sentry.qtopay.net/share/issue/1e32005a46e94351b1bcfaa2feae6352/)

```
//swoole_websocket.php
/*
    |--------------------------------------------------------------------------
    | Default websocket driver
    |--------------------------------------------------------------------------
    */
    'default' => 'redis',
```

log error

Symfony\Component\Debug\Exception\FatalThrowableError: Argument 2 passed to SwooleTW\Http\Websocket\Rooms\RedisRoom::removeValue() must be of the type array, int given, called in /mnt/data/wwwroot/qtopay.net/vendor/swooletw/laravel-swoole/src/Websocket/Rooms/RedisRoom.php on line 121
#3 vendor/swooletw/laravel-swoole/src/Websocket/Rooms/RedisRoom.php(160): removeValue
#2 vendor/swooletw/laravel-swoole/src/Websocket/Rooms/RedisRoom.php(121): delete
#1 vendor/swooletw/laravel-swoole/src/Websocket/Websocket.php(150): leave
#0 vendor/swooletw/laravel-swoole/src/Concerns/InteractsWithWebsocket.php(174): null

5. What did you expect to see?

6. What did you see instead?
i have no idea, mybe i will use table for default websocket driver

**Comment by albertcht** [comment]
Hi @zhongjq ,

I'm not able to reproduce this issue on my local environment. And I it weird that this shouldn't happen in this function:

```
public function delete(int $fd, $rooms)
{
    $rooms = is_array($rooms) ? $rooms : [$rooms];
    $rooms = count($rooms) ? $rooms : $this->getRooms($fd);

    $this->removeValue($fd, $rooms, RoomContract::DESCRIPTORS_KEY);
    // ...
```

The first `$rooms` variable must be array type. The second `$rooms` must be array as well because `$this->getRooms($fd)` calls `smembers` function from predis, and it must return an array.

Comment below if anyone finds other possible clues.

**Comment by zhongjq** [comment]
Hi @albertcht 
Yes, this is my low probability event in the production environment so I doubt it will happen when high concurrency is online. 
I have also studied the code and found no logic problems. This error does not affect the operation of my online website, so it is not easy to locate

**Comment by liyu001989** [comment]
same issue with driver error , low probability. I trid to logger `$rooms` , the value  is  1。

**Comment by albertcht** [comment]
@liyu001989 , thanks for your info, I will try to figure it out yet no clues now.

**Comment by liyu001989** [comment]
I add some logs in `Websocket/Rooms/RedisRoom.php`。

![image](https://user-images.githubusercontent.com/2981799/63140093-3c4b7300-c013-11e9-9642-8a3153849aa7.png)

sometimes redis  smembers return  `1` or  `0`。

![image](https://user-images.githubusercontent.com/2981799/63140105-479e9e80-c013-11e9-9b4a-92bacb674bb7.png)
![image](https://user-images.githubusercontent.com/2981799/63141924-1bd2e700-c01a-11e9-9ede-7afb8a42d926.png)


In redis-cli or tinker  smembers is ok

![image](https://user-images.githubusercontent.com/2981799/63140165-80d70e80-c013-11e9-9377-262efac4a78b.png)
![image](https://user-images.githubusercontent.com/2981799/63140170-86345900-c013-11e9-8ccd-ce39a8d34ca0.png)

the smembers function not always returning array，any idea ？

**Comment by albertcht** [comment]
Hi @liyu001989 , thanks for your report, I will keep digging on it.

**Comment by mxp100** [comment]
problem in this code:
`    /**
     * @param \Predis\Client|null $redis
     *
     * @return \SwooleTW\Http\Websocket\Rooms\RoomContract
     */
    public function prepare(RedisClient $redis = null): RoomContract
    {
        $this->setRedis($redis);
        $this->setPrefix();
//        $this->cleanRooms();

        return $this;
    }`

**Comment by killtw** [comment]
It would be found when unsuspected disconnect happen

**Comment by albertcht** [comment]
I try to fix that error in the release `v2.6.64`. Can you try to keep watching on this issue?
